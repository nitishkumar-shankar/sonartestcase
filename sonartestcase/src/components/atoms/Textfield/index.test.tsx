import { render, screen } from "@testing-library/react";
import Input from '../Textfield/index'
import { describe, expect, it } from "vitest";

describe("Input Atom", () => {
  it("renders with the correct placeholder", () => {
    render(
      <Input placeholder="Enter your name" value="" onChange={() => {}} />
    );
    const input = screen.getByPlaceholderText("Enter your name");
    expect(input).toBeInTheDocument();
  });

  it("displays the correct value", () => {
    render(
      <Input
        placeholder="Enter your name"
        value="John Doe"
        onChange={() => {}}
      />
    );
    const input = screen.getByDisplayValue("John Doe");
    expect(input).toBeInTheDocument();
  });

  it("matches the snapshot", () => {
    const { asFragment } = render(
      <Input placeholder="Enter your name" value="" onChange={() => {}} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});