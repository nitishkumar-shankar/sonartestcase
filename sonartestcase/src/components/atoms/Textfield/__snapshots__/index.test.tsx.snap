// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Input Atom > matches the snapshot 1`] = `
<DocumentFragment>
  <input
    data-testid="custom-input"
    placeholder="Enter your name"
    type="text"
    value=""
  />
</DocumentFragment>
`;
